spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java

```diff
@@ -880,14 +880,10 @@ boolean cacheReturner(CloseSafeProducer<K, V> producerToRemove, Duration timeout
 		else {
 			this.globalLock.lock();
 			try {
-				if (producerToRemove.epoch != this.epoch.get()) {
-					producerToRemove.closeDelegate(timeout, this.listeners);
-					return true;
-				}
```

Comment:
Have a look at https://github.com/spring-projects/spring-kafka/blob/2306ac5cbd47702f9a492541186944682edada9c/spring-kafka/src/main/java/org/springframework/kafka/core/DefaultKafkaProducerFactory.java#L762-L788

At line L763-L765, create tx producer is diff from nomal producer,
tx producer is not managed by `DefaultKafkaProducerFactory`, so producer suffix not release when call `DefaultKafkaProducerFactory#reset()` if producer is in transaction. (eg. use tx producer in `containerListener`)

Maybe recover this code is better, i'm not sure, please help take a look, @sobychacko @artembilan